catalog/spatial/csw/spatial-csw-transformer/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/TestXstreamPathConverter.java

```diff
@@ -86,6 +86,46 @@ public void setup() {
     }
 
     @Test
+    public void testRepeatedElements() throws XMLStreamException {
+        String xml = "<gml:Polygon xmlns:gml=\"http://www.opengis.net/gml\">\n" + "<gml:B>\n"
+                + "<gml:C>\n" + "<gml:D>\n" + "<gml:E>\n" + "<gml:F>\n" + "<gml:G>value1</gml:G>\n"
+                + "</gml:F>\n" + "<gml:F>\n" + "<gml:G>value2</gml:G>\n" + "</gml:F>\n"
+                + "</gml:E>\n" + "</gml:D>\n" + "<gml:D>\n" + "<gml:E>\n" + "<gml:F>\n"
+                + "<gml:G>value3</gml:G>\n" + "</gml:F>\n" + "<gml:F>\n" + "<gml:G>value4</gml:G>\n"
+                + "</gml:F>\n" + "</gml:E>\n" + "</gml:D>\n" + "</gml:C>\n" + "<gml:C>\n"
+                + "<gml:D>\n" + "<gml:E>\n" + "<gml:F>\n" + "<gml:G>value5</gml:G>\n" + "</gml:F>\n"
+                + "<gml:F>\n" + "<gml:G>value6</gml:G>\n" + "</gml:F>\n" + "</gml:E>\n"
+                + "</gml:D>\n" + "<gml:D>\n" + "<gml:E>\n" + "<gml:F>\n" + "<gml:G>value7</gml:G>\n"
+                + "</gml:F>\n" + "<gml:F>\n" + "<gml:G>value8</gml:G>\n" + "</gml:F>\n"
+                + "</gml:E>\n" + "</gml:D>\n" + "</gml:C>\n" + "</gml:B>\n" + "</gml:Polygon>";
+
+        Path path = new Path("/Polygon/B/C/D/E/F/G");
+
+        Set<Path> paths = new LinkedHashSet<>();
+        paths.add(path);
+        argumentHolder.put(XstreamPathConverter.PATH_KEY, paths);
+
+        XMLStreamReader streamReader = XMLInputFactory.newInstance()
+                .createXMLStreamReader(new StringReader(xml));
+        HierarchicalStreamReader reader = new StaxReader(new QNameMap(), streamReader);
+        XstreamPathValueTracker pathValueTracker = (XstreamPathValueTracker) xstream.unmarshal(
+                reader,
+                null,
+                argumentHolder);
+
+        assertThat(pathValueTracker.getAllValues(path),
```

Comment:
✏️ You should probably add some negative test paths as well, where the nesting is incorrect, to ensure you don't return those values.